### DISCUSSION TITLE
How to use Invoke-VenafiRestMethod to update custom field

### DISCUSSION THREAD

**Discussion by tonyb2BB** [Venafi/venafips#91] [Q&A] [answered]
Looking to use VenafiPS to configure a custom field's list options. Basically want a custom field thats pulling a list from another source so will be somewhat dynamic as in adding a new option from time to time. For testing I have a new custom field named 'TestField' with 3 options. As I didn't see an Update Custom Field built-in functionality I'm trying to use the Invoke-VenafiRestAPI method to interact with the Metadata/UpdateItem endpoint as follows:
Invoke-VenafiRestMethod -Method Post -UriLeaf 'Metadata/UpdateItem'  -Body @{'Update' = @{'DN' = '\VED\Metadata Root\TestField'; 'Data' = @{'Name' = 'ErrorMessage'; 'Value' = 'Test Help Message'}}}. 

I've also tried taking it as pure JSON and then converting it to a PS Hash Table.
$json = '{
>>    "Update":{
>>       "DN":"\\VED\\Metadata Root\\TestField",
>>       "Data": [
>>          {
>>             "Name":"Help",
>>             "Value":"Test Help Message"
>>          }
>>   ]
>>    }
>> }'
>> $hashtable = $json | ConvertFrom-Json | ConvertTo-HashTable
>> Invoke-VenafiRestMethod -Method Post -UriLeaf 'Metadata/UpdateItem'  -Body $hashtable

Both methods I end up with the same error:
Error                                                         Locked Result
-----                                                         ------ ------
Metadata item has invalid values. The update Data is missing.  False     15

If I look at my $hashtable, it appears to have the Data field added correctly.
>>  $hashtable.Update.Data
Name                           Value
----                           -----
Value                          Test Help Message
Name                           Help

**Comment by gdbarron** [comment, marked as answer]
Hi @tonyb2BB, you were close, but I see 2 issues.  The first code, `Invoke-VenafiRestMethod -Method Post -UriLeaf 'Metadata/UpdateItem' -Body @{'Update' = @{'DN' = '\VED\Metadata Root\TestField'; 'Data' = @{'Name' = 'ErrorMessage'; 'Value' = 'Test Help Message'}}}` has Data as a hashtable when it should be an array of hashtables (it looks like you fixed this in your second try).  Also, a list type doesn't have an error message, only a string does when performing validation.  When I try and update ErrorMessage for a list field I get a 400.

The following worked for me to update list options, `Invoke-VenafiRestMethod -UriLeaf 'Metadata/UpdateItem' -Body @{'Update'=@{'DN'='\VED\Metadata Root\bb163985-c9a6-472c-ae2c-dc2f58e591f6';'Data'=@(@{'Name'='AllowedValues';'Value'=@('item1','item2')})}} -Method Post`

**Reply by tonyb2BB** [reply]
Thanks! Got pretty close to this with further testing but still didn't have my hashtable array formatted correctly.
